### PR TITLE
blockSubscribe RPC Method

### PR DESCRIPTION
**Pull request by ruiqic**

### CODE REVIEW

**Comment by michaelhly** [comment]
Hi, thanks for opening a PR! Would you mind fixing the lint issue and also writing a test?

You can run `make lint` to run the linter, and `make format` to format your code. Here is an example of a test:
https://github.com/michaelhly/solana-py/blob/53adfce3b5b8d284826658764b434628f70083d7/tests/integration/test_websockets.py#L283-L292

**Comment by ruiqic** [comment]
Hi, thanks for checking. I just added a simple test for slot and used `make format` to fix all `black` issues. Let me know.

**Comment by ruiqic** [comment]
Test not passing maybe because not all RPCs support this method:

> UNSTABLE METHOD
This subscription is considered unstable and is only available if the validator was started with the --rpc-pubsub-enable-block-subscription flag. The format of this subscription may change in the future.
(https://solana.com/docs/rpc/websocket/blocksubscribe)

**Comment by michaelhly** [comment]
Hi @ruiqic. You can add the flag here:
https://github.com/michaelhly/solana-py/blob/53adfce3b5b8d284826658764b434628f70083d7/tests/docker-compose.yml#L11

**Comment by michaelhly** [comment]
Also, would you mind reverting all the formatting changes not related to this PR?

File: src/solana/rpc/websocket_api.py [hunk 1]
```diff
@@ -176,6 +183,52 @@ async def logs_unsubscribe(
...
+            filter_: filter criteria for the blocks.
+            commitment: The commitment level to use.
+            encoding: Encoding to use.
+            transaction_details: level of transaction detail to return.
+            show_rewards: whether to populate the rewards array.
+            max_supported_transaction_version: the max transaction version to return in responses.
+        """
+        req_id = self.increment_counter_and_get_id()
+        commitment_to_use = None if commitment is None else _COMMITMENT_TO_SOLDERS[commitment]
+        encoding_to_use = None if encoding is None else _ACCOUNT_ENCODING_TO_SOLDERS[encoding]
```
Comment (michaelhly) [inline src/solana/rpc/websocket_api.py:207]: Looks like we need to use `_TX_ENCODING_TO_SOLDERS` instead:
```suggestion
        encoding_to_use = None if encoding is None else _TX_ENCODING_TO_SOLDERS[encoding]
```

https://github.com/ruiqic/solana-py/blob/4d8be0a047c7be16dea1058f2b6f9d321bea9fe5/src/solana/rpc/core.py#L98-L104

**Comment by ruiqic** [comment]
Hi, thanks for checking the code. How can the coverage test be fixed?

**Comment by michaelhly** [comment]
I'll fix that separately. @ruiqic, would you mind giving me push access to your forked repo? I just want to push some clean up.

**Review by michaelhly — APPROVED** [review]
LGTM! Thanks!

**Comment by ruiqic** [comment]
Thanks. I already had `Allowing edits by maintainers` checked but I invited you as collaborator too.

**Pull request closed without merging**